### PR TITLE
Var dicts

### PR DESCRIPTION
**Pull request by nguy**
This PR is not complete yet. I want to get feedback. When changing the readers to dictionaries, it is required to modify the ```DropSizeDistribution``` object class. 

@josephhardinee can you:
1. Verify the changes submitted work, i.e. calculations perform as expected
2. Should we save spread, bin_edges, diameter as dictionaries or just arrays?

All **aux** readers will need updated.

### CODE REVIEW

**Comment by nguy** [comment]
ping @josephhardinee Let me know when you get to this so I can adjust code accordingly and update other needed areas.

File: pydisdrometer/io/JWDReader.py [hunk 1]
```diff
@@ -93,8 +95,37 @@ def _read_file(self):
...
-        self.Nd = np.ma.array(self.Nd)
+        fields = {}
+##        self.Nd = np.ma.array(self.Nd)
+##        self.rain_rate = np.ma.array(self.rain_rate)
         self.time = np.ma.array(self.time)
         self.time = self.time - self.time[0]
-        self.rain_rate = np.ma.array(self.rain_rate)
+
+        self.fields['Nd'] = common.var_to_dict(
+            'Nd', np.ma.array(self.Nd), 'm^-3',
```
Comment (josephhardinee) [inline pydisdrometer/io/JWDReader.py]: The units should be 'm^-3mm^-1.

**Comment by josephhardinee** [comment]
1) It all appears to work on mine. 
2) I think probably as dicts to remain consistent. 

I'll add comments to the diffset. I saw a few minor things but all in all it appears to be pretty good!

File: pydisdrometer/io/ParsivelNasaGVReader.py [hunk 2]
```diff
@@ -98,22 +98,31 @@ def __init__(self, filename, campaign, skip_header):
...
+
         self.bin_edges = np.hstack(
             (0, self.diameter + np.array(self.spread) / 2))
 
         self.f.close()
 
+    def _prep_data(self):
+        self.fields = {}
+        self.fields['Nd'] = common.var_to_dict(
+            'Nd', np.ma.array(self.Nd), 'm^-3',
```
Comment (josephhardinee) [inline pydisdrometer/io/ParsivelNasaGVReader.py]: Same with the units here(add the per millimeter)

File: pydisdrometer/io/ParsivelReader.py [hunk 3]
```diff
@@ -17,10 +20,11 @@ def read_parsivel(filename):
     '''
 
     reader = ParsivelReader(filename)
-    dsd = DropSizeDistribution(reader.time, reader.Nd, reader.spread,
-                               rain_rate=reader.rain_rate, velocity=reader.velocity,
-                               Z=reader.Z, num_particles=reader.num_particles,
-                               bin_edges=reader.bin_edges, diameter=reader.diameter)
+##    dsd = DropSizeDistribution(reader.time, reader.Nd, reader.spread,
```
Comment (josephhardinee) [inline pydisdrometer/io/ParsivelReader.py]: You can go ahead and delete the commented out lines.

File: pydisdrometer/io/JWDReader.py (hunk 1 above)
Comment (nguy) [inline pydisdrometer/io/JWDReader.py]: Oops, nice catch. Fixed.

File: pydisdrometer/io/ParsivelNasaGVReader.py (hunk 2 above)
Comment (nguy) [inline pydisdrometer/io/ParsivelNasaGVReader.py]: Done.

**Comment by nguy** [comment]
Are spread, bin_edges, diameter in millimeters?

**Comment by josephhardinee** [comment]
Yes, they should be. A few file formats have them in something else and I try to make sure to always convert them to mm. Just makes life easier.

**Comment by nguy** [comment]
Cool, I wanted to verify. So I'll start making more changes and push them up here for you to review also before merger.

**Comment by nguy** [comment]
@josephhardinee I added dictionary for spread, diameter, bin_edges. Converted the entire "io" code base. Can you check again real fast to see if I didn't break anything?

**Comment by nguy** [comment]
Most of the readers in the the ```aux_readers``` seciton need work on the time variable to calculate time. Specifically need to grab the date information of the file.

File: pydisdrometer/io/JWDReader.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 import numpy as np
+from netCDF4 import num2date, date2num
+
```
Comment (josephhardinee) [inline pydisdrometer/io/JWDReader.py]: This needs a from .io import common

File: pydisdrometer/aux_readers/GPMApuWallopsRawReader.py [hunk 5]
```diff
@@ -1,11 +1,7 @@
 # -*- coding: utf-8 -*-
 import numpy as np
-import numpy.ma as ma
 from ..DropSizeDistribution import DropSizeDistribution
```
Comment (josephhardinee) [inline pydisdrometer/aux_readers/GPMApuWallopsRawReader.py]: This needs a  from ..io import common

Comment (josephhardinee) [inline pydisdrometer/aux_readers/GPMApuWallopsRawReader.py]: This needs a from ..io import common

File: pydisdrometer/io/ParsivelNasaGVReader.py [hunk 6]
```diff
@@ -98,22 +92,34 @@ def __init__(self, filename, campaign, skip_header):
                 self.time.append(self._parse_time(map(int, (row[0].split()[0:4]))))
                 self.Nd.append([float(x) for x in row[0].split()[4:]])
 
+        self._prep_data()
+
+        self.bin_edges = common.var_to_dict(
+            'bin_edges',
+            np.hstack(
+                (0, self.diameter + np.array(self.spread) / 2)),
```
Comment (josephhardinee) [inline pydisdrometer/io/ParsivelNasaGVReader.py]: I think this should be self.diameter['data']

**Comment by nguy** [comment]
I think I addressed all of these. Good catches!

**Comment by nguy** [comment]
Should we merge this or fix the time issues in the aux readers?

**Pull request merged**